server/src/main/java/org/apache/cassandra/sidecar/config/yaml/InstanceConfigurationImpl.java

```diff
@@ -21,80 +21,63 @@
 import java.util.Collections;
 import java.util.List;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import org.apache.cassandra.sidecar.config.InstanceConfiguration;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * Encapsulates the basic configuration needed to connect to a single Cassandra instance
  */
 public class InstanceConfigurationImpl implements InstanceConfiguration
 {
-    @JsonProperty("id")
     protected final int id;
-
-    @JsonProperty("host")
     protected final String host;
-
-    @JsonProperty("port")
     protected final int port;
-
-    @JsonProperty("data_dirs")
+    protected final String cassandraHomeDir;
     protected final List<String> dataDirs;
-
-    @JsonProperty("staging_dir")
     protected final String stagingDir;
-
-    @JsonProperty("cdc_dir")
     protected final String cdcDir;
-
-    @JsonProperty("jmx_host")
+    protected final String commitlogDir;
+    protected final String hintsDir;
+    protected final String savedCachesDir;
+    protected final String localSystemDataFileDir;
     protected final String jmxHost;
-
-    @JsonProperty("jmx_port")
     protected final int jmxPort;
-
-    @JsonProperty("jmx_ssl_enabled")
     protected final boolean jmxSslEnabled;
-
-    @JsonProperty("jmx_role")
     protected final String jmxRole;
-
-    @JsonProperty("jmx_role_password")
     protected final String jmxRolePassword;
 
-    public InstanceConfigurationImpl()
```

Comment:
Removed this in order to reduce the confusion with what can be null and what cannot be null.